hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/DeleteBlocksCommandHandler.java

```diff
@@ -265,6 +271,46 @@ public void run() {
     }
   }
 
+  /**
+   * A cache to store committed transaction IDs with a fixed maximum size,
+   * ensuring the oldest entries are evicted when capacity is exceeded.
+   */
+  public static final class CommittedTransactionCache {
+    private final Set<Long> committedTransactionIds;
+    private final int maxCacheSize;
+
+    public CommittedTransactionCache(int maxCacheSize) {
+      this.committedTransactionIds = new LinkedHashSet<>(maxCacheSize);
+      this.maxCacheSize = maxCacheSize;
+    }
+
+    public boolean contains(Long txID) {
+      return committedTransactionIds.contains(txID);
+    }
+
+    public int getCacheSize() {
+      return committedTransactionIds.size();
+    }
+
+    public void adds(List<Long> txIDs) {
+      committedTransactionIds.addAll(txIDs);
+      ensureCapacity();
+    }
+
+    private void ensureCapacity() {
+      int size = committedTransactionIds.size();
+      if (size > maxCacheSize) {
+        int numToRemove = size - maxCacheSize;
+        Iterator<Long> it = committedTransactionIds.iterator();
+        while (numToRemove > 0 && it.hasNext()) {
```

Comment:
So it is a FIFO cache. @xichen01 , do you have any idea about the duplicate pattern?  Which policy will be more suitable,  FIFO or LRU?